pkg/steps/testing.go

```diff
@@ -37,13 +76,97 @@ func (dl *DryLogger) AddObject(o runtime.Object) {
 	dl.objects = append(dl.objects, o)
 }
 
-// GetObjects returns the list of objects.
+func (dl *DryLogger) AddBuild(build *buildapi.Build) {
```

Comment:
nit: you used to sort the deep copy, now you sort the object someone hands you -- do the deep copy first

Reply:
`DeepCopyObject()` is converting the build to a `runtime.Object`

Reply:
We can still cast it back safely, it's very surprising for a logger to mutate the objects.